onnxruntime/core/graph/graph.cc

```diff
@@ -1933,6 +1949,38 @@ void SortForwardNodesByReverseDFS(const Graph* graph,
   // Note 2: While it is also possible some nodes not contributing to the forward output nodes will be
   //   executed before YieldOp, for example, if one forward node's output is used by Shape/Size, then
   //   the Shape/Size node should be executed before YieldOp to release the memory as soon as possible.
+  InlinedVector<size_t> nodes_in_degree;
+  std::queue<const Node*> to_visit;
```

Comment:
(I'm not sure about this) Since you've added the 'distance-to-input_leaf' concept now, would it be better to use a priority_queue here with the NodeCompareByMaxDistance function for comparison?

Reply:
It may work in some cases, while still possible we get a bad ordering for example, if we have few input leaf nodes:
Transpose(weight_layer_11), Transpose(weight_layer_10), ..., Transpose(weight_layer_1), Transpose(weight_layer_0), ATen(embedding). 

If we accidentally start with layer_11's Transpose, then its children; then layer_10's Transpose, ..., layer_0's Transpose; at last Aten(embedding). Then you would see all weights got transposed before the node that need immediate execution get deferred. 

Using the max_distance together with Reversed DFS from forward outputs, when it goes to the nodes that taking layer_11's Transpose, and another node N (generated by layer_10) as inputs, node N will be preferred to be output first because its distance to graph input is bigger than layer_11's Transpose.

Reply:
Make sense, thank you!